packages/notifi-frontend-client/lib/client/ensureSource.ts

```diff
@@ -671,34 +671,25 @@ const getXMTPFilter = (
   };
 };
 
-type HealthCheckEventInputsWithIndex = {
+export type HealthCheckEventInputsWithIndex = {
   index: number; // The index of CheckRatio list
   [key: string]: unknown; // The rest of inputs
 };
 
-type HealthCheckEventInputsWithCustomPercentage = {
+export type HealthCheckEventInputsWithCustomPercentage = {
   thresholdDirection: ThresholdDirection; // The direction of threshold
   customPercentage: number; // The percentage value of custom health check
   [key: string]: unknown; // The rest of inputs
 };
 
-type HealthCheckInputs =
+export type HealthCheckInputs =
   | HealthCheckEventInputsWithIndex
   | HealthCheckEventInputsWithCustomPercentage;
 
-const healthCheckInputsValidator = (
-  inputs: Record<string, unknown>,
-): inputs is HealthCheckInputs => {
-  if ('index' in inputs || 'customPercentage' in inputs) {
-    return true;
-  }
-  return false;
-};
-
 const getHealthCheckFilter = (
   source: Types.SourceFragmentFragment,
   eventType: HealthCheckEventTypeItem,
-  inputs: HealthCheckInputs,
+  inputs: HealthCheckInputs | Record<string, unknown>,
```

Comment:
let's just make this `Record<string, unknown>` since we do type checking inside the function body

Reply:
Issue Address!